pandas/tests/io/test_parquet.py

```diff
@@ -473,6 +473,18 @@ def test_partition_cols_supported(self, pa, df_full):
             assert len(dataset.partitions.partition_names) == 2
             assert dataset.partitions.partition_names == set(partition_cols)
 
+    def test_partition_cols_string(self, pa, df_full):
+        # GH #23283
+        partition_cols = "bool"
+        df = df_full
+        with tm.ensure_clean_dir() as path:
+            df.to_parquet(path, partition_cols=partition_cols, compression=None)
+            import pyarrow.parquet as pq
+
+            dataset = pq.ParquetDataset(path, validate_schema=False)
+            assert len(dataset.partitions.partition_names) == 1
+            assert dataset.partitions.partition_names == set((partition_cols))
```

Comment:
Seems I misunderstood the `set` API - it's splitting up the string into a character set. 

Try changing the line to:
```assert dataset.partitions.partition_names == [partition_cols]```

Not sure if that's a pep8 error but run the tests/black locally before updating the PR to make sure

Reply:
I changed it then ran the tests and it failed with AssertionError: assert {'bool'} == ['bool']. I have tried a couple different things and can't really find which one is right.

Reply:
I added a partition_cols_list variable that takes the partition_cols and turns it into a list then use that instead like so 

partition_cols_list = [partition_cols]
...
assert dataset.partitions.partition_names == set(partition_cols_list)

This passes the local black and the tests